etp-db/src/main/sql/migration/repeatable/r-laatija.sql

```diff
@@ -1,11 +1,11 @@
 create or replace function patevyys_paattymisaika(l laatija) returns timestamp as $$
 begin
-  return timezone('Europe/Helsinki', l.toteamispaivamaara::timestamp) + interval '7 year' + interval '1 day';
+  return timezone('Europe/Helsinki', l.toteamispaivamaara::timestamp + interval '7 year' + interval '1 day');
```

Comment:
timezone-funktio palauttaa lokaalin ajan ja tekee aikavyöhykkeen muutoksen, joten kaikki aikaan liittyvä muutos pitää tehdä ennen funktion kutsumista.

Reply:
SELECT timezone('Europe/Helsinki', '2021-03-25'::timestamp); 
 => 2021-03-24 22:00:00+00. Oikein: kelloa ei ole vielä siirretty eteenpäin.
 
 SELECT timezone('Europe/Helsinki', '2021-03-25'::timestamp) + interval '5 days';
 => 2021-03-29 22:00:00+00. Tämä on "väärin", sillä 27.3.2021 kelloja siirrettiin.
 
 SELECT timezone('Europe/Helsinki', '2021-03-25'::timestamp + interval '5 days');
 => 2021-03-29 21:00:00+00, oikein. Sulkumerkkiä siirretty, eli nyt otetaan valmiiksi oikea päivä ennen aikavyöhykkeen huomioimista.